### PR TITLE
V1.4.1 update dg

### PR DESCRIPTION
**Pull request by marcustw**
1. Update DG
2. Added AtasParser javadoc

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -499,23 +508,42 @@ todo
...
+* Pros: 
+   * Easy to implement. One example is through the usage of Java Cryptography API.
+
+* Cons: 
+   * Performance reduction may occur since every call to save to storage requires encrypting a lot of data.
+   * Security issues may still arise if we store the encryption key in the same machine.
 
-{ end of `implementation#dat_encryption` written by: ______________ }
+### 4.10.2 Solution 2
+Store data outside the user’s machine and issue them access token.
```
Comment (nweiyue) [inline docs/DeveloperGuide.md]: Perhaps you can change "them" to "him/her" or change "user's to users'"?

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -499,23 +508,42 @@ todo
...
+
+* Pros: 
+   * Data will not be only bounded to one machine, hence, can be restored if the user forgets their credentials.
```
Comment (nweiyue) [inline docs/DeveloperGuide.md]: Perhaps "Data will be bounded outside of one machine/bounded to other machines or servers" may flow better?
Again, should it be either "user" and "his/her" or "users" and "their"?

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -771,25 +799,44 @@ testers are expected to do more *exploratory* testing.
...
+   One or more (but less than 100) student is shown in the StudentListPanel.
+   
+   1. Test case: `deletestu 1`<br>
+      Expected: First contact is deleted from the list. Details of the deleted contact shown in the ResultDisplay.
+   
+   1. Test case: `deletestu 0`<br>
+      Expected: No student is deleted. Error details shown in the ResultDisplay. 
+   
+   1. Test case: `deletestu 101`<br>
+      Expected: No student is deleted. Error details shown in the ResultDisplay.
```
Comment (nweiyue) [inline docs/DeveloperGuide.md:1270]: Perhaps this can be "Similar to previous."?

File: docs/UserGuide.md [hunk 4]
```diff
@@ -72,13 +72,14 @@ Specifically, **ATAS** allows you to keep track of four different aspects (see [
 ### 1.3. Preview
 
 The following is the GUI of our application:
-    ![UiPreview](images/UiPreview.png)
+    ![UiPreview](images/user-guide/1.3-UiPreview.png)
+    <p align="center"> <sub> <b>Figure 1.3</b>: UI Preview </sub> </p>
```
Comment (nweiyue) [inline docs/UserGuide.md]: Should this be "1.3.1"?

**Review by nweiyue — APPROVED** [review]
LGTM! Just some nits.

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (marcustw) [reply docs/DeveloperGuide.md]: Thank you for the suggestion!

File: docs/UserGuide.md (hunk 4 above)
Comment (marcustw) [reply docs/UserGuide.md]: I think it should be "Figure 1.3-1".

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (marcustw) [reply docs/DeveloperGuide.md:1270]: Should I? I'm afraid it wouldn't be clear enough which previous I am referring to.

Comment (nweiyue) [reply docs/DeveloperGuide.md:1270]: I see. I think this should be fine as well 👍

**Review by nweiyue — APPROVED** [review]
LGTM!

**Pull request merged**